packages/hash/frontend/src/components/layout/PageHeader/SearchBar/index.tsx

```diff
@@ -95,38 +129,56 @@ const SearchBarWhenSearchIsEnabled: React.VFC = () => {
 
   useKey(["Escape"], () => setResultListVisible(false));
 
-  useKeys(["AltLeft", "KeyK"], () => inputRef.current?.focus());
-
   const [rootRef] = useOutsideClickRef(() => setResultListVisible(false));
 
   // present loading screen while waiting for the user to stop typing
   const isLoading = loading || displayedQuery !== submittedQuery;
 
   return (
-    <div ref={rootRef} className={tw`relative h-full w-full`}>
-      <div className={tw`absolute h-full flex flex-row items-center`}>
-        <SearchIcon className={tw`m-2 scale-150`} />
-      </div>
-      <input
-        className={tw`p-2 pl-10 w-1/2 border border-gray-200 rounded-lg focus:outline-none`}
-        ref={inputRef}
-        placeholder="Search (Alt+k)"
-        type="text"
-        value={displayedQuery}
-        onFocus={() => setResultListVisible(true)}
-        onChange={(event) => {
-          setResultListVisible(true);
-          setQueryText(event.target.value);
-        }}
-      />
+    <Box
+      sx={{
+        marginLeft: 0,
+        [theme.breakpoints.up("md")]: {
+          marginLeft: theme.spacing(3),
+        },
+        position: "relative",
+        height: "100%",
+        ...(isMobile && displaySearchInput
+          ? {
+              position: "absolute",
+              width: "100%",
+              zIndex: 1,
+              left: 0,
+              top: theme.spacing(1.5),
+              px: 2,
+            }
+          : {}),
+      }}
+      ref={rootRef}
+    >
+      {!isMobile ? (
+        <DesktopSearch
+          displayedQuery={displayedQuery}
+          setQueryText={setQueryText}
+          setResultListVisible={setResultListVisible}
+        />
+      ) : (
+        <MobileSearch
+          displayedQuery={displayedQuery}
+          setQueryText={setQueryText}
+          setResultListVisible={setResultListVisible}
+          displaySearchInput={displaySearchInput}
+          setDisplaySearchInput={setDisplaySearchInput}
+        />
+      )}
```

Comment:
I'm unconvinced these should be separate components. There's significant overlap in their functionality and styling, could be not just use a `isMobile` boolean to add the necessary styling/functionality that is dependent on whether the user is on mobile or on desktop?

Reply:
(Given it may be time-consuming to refactor this, I'm happy for it to be addressed in a follow-up PR)